test/validators.js

```diff
@@ -828,6 +831,32 @@ describe('Validators', () => {
       ],
     });
   });
+  it('should invalidate FQDN when not require_tld', () => {
+    test({
+      validator: 'isFQDN',
+      args: [
+        { require_tld: false },
+      ],
+      invalid: [
+        'example.0',
+        '192.168.0',
+        '192.168.0.9999',
+      ],
+    });
+  });
+  it('should validate FQDN when not require_tld but allow_numeric_tld', () => {
+    test({
+      validator: 'isFQDN',
+      args: [
+        { allow_numeric_tld: true, require_tld: false },
+      ],
+      valid: [
+        'example.0',
```

Comment:
Why would `example.0` be a valid FQDN? :thinking: And even `.9999`, is there anything I'm not understanding...

Reply:
Hi @profnandaa ! Thanks for the quick response.

> Why would `example.0` be a valid FQDN? And even .9999, is there anything I'm not understanding...

In this test case, `allow_numeric_tld` is `true` and `require_tld` is `false`, this mimics the current behavior of `isFQDN` when `require_tld` is `false`. In that way, `isFQDN` will allow numeric TLDs. 

I think this solution make sense and doesn't complicate the implementation of `isFQDN`. 

#### Other options?

The [RFC 3696](https://tools.ietf.org/html/rfc3696#section-2) mentions two ways to validate the TLD:

One is by polling the list published by IANA [DomainList](http://data.iana.org/TLD/tlds-alpha-by-domain.txt) and validate against it. However, based on the [comment in #704](https://github.com/validatorjs/validator.js/issues/704#issuecomment-325167598):

> Ideally the validator would check TLDs against a list of known values, however the addition of gTLDs mean that the list would be long and ever-growing. I'm not interested in bloating the library or keeping the list up to date.

Since we don't want to keep that list up today, the second option is to validate that the TLD name is not all-numeric.

Does it make sense?

#### Note:
I think we could avoid the parameter `allow_numeric_tld` and **always** require that the TLD is not all-numeric. What do you think?

Reply:
Makes sense. Really sorry for the long delay on this one, can't believe it has almost been a year!

/cc. @tux-tn @ezkemboi -- could you please have a look at this and advise.

Reply:
It makes sense to me @profnandaa. 
After checking on comments and some research online, I agree with the changes.